yt_dlp/extractor/vidly.py

```diff
@@ -0,0 +1,76 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    mimetype2ext,
+    url_or_none,
+)
+from ..utils.traversal import traverse_obj
+
+
+class VidlyIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:vid\.ly/|(?:s\.)?vid\.ly/embeded.html\?(?:[^#/]+&)?link=)(?P<id>\w+)'
+    _EMBED_REGEX = [r'<script[^>]+\bsrc=[\'"](?P<url>(?:https?:)?//vid\.ly/\w+/embed[^\'"]+)',
+                    r'<iframe[^>]+\bsrc=[\'"](?P<url>(?:https?:)?//(?:s\.)?vid\.ly/embeded.html\?(?:[^#/]+&)?link=\w+[^\'"]+)']
```

Comment:
do you have any `_WEBPAGE_TESTS` we can add?

Reply:
> I was unable to find a single site using vid.ly for _WEBPAGE_TESTS. Maybe someone else has better luck looking for those.

Reply:
Okay found one https://www.petfinder.com/dog/gus-57378930/tn/ooltewah/furever-furkids-rescue-tn592/